interpmike/animeal_android
# Feature/EPMEDU-1850 Show dialog when another user has booked the feeding

[Ticket reference](https://jira.epam.com/jira/browse/EPMEDU-1850)

Adds logic to launch dialog when another user already booked feeding. 

Gets result from Feeding Handler and when the reason for Dismiss state is Request Cancelled, it will show warning once. This change was added on Favourites, Search and Home screens (screenshot attached).

Steps to validate:

- Use 2 devices each one with a different account.
- Select same feeding point at favorites/Home/Search screen on both devices.
- Go to confirmation feeding screen on both.
<img src="https://github.com/AnimealProject/animeal_android/assets/4513325/393e71eb-4838-4a8a-9180-ddb67b984442" width="75" height="155">

- Confirm feeding (Agree button) at Device 1 only and wait till route to feeding appears.
- Confirm feeding at Device 2, feeding warning dialog appears.

<img src="https://github.com/AnimealProject/animeal_android/assets/4513325/1b0e7a31-af6e-43e8-97bd-60941af504e2" width="150" height="310">

<img src="https://github.com/AnimealProject/animeal_android/assets/4513325/f32a0dfc-1277-4ca5-a2e0-c74b452cef03" width="150" height="310">

<img src="https://github.com/AnimealProject/animeal_android/assets/4513325/5a2ed61a-c4cf-4d6d-a9f1-9873a093984f" width="150" height="310">
